test/flash.test.js

```diff
@@ -343,6 +343,54 @@ QUnit.test('drops tags before currentTime when seeking', function() {
             'three tags are appended');
 });
 
+QUnit.test('drops audio tags before the buffered end always', function() {
+  let sourceBuffer = this.mediaSource.addSourceBuffer('video/mp2t');
+  let i = 10;
+  let endTime;
+  let tags_ = [];
+
+  this.mediaSource.tech_.buffered = function() {
+    return videojs.createTimeRange([[0, endTime]]);
+  };
+
+  // push a tag into the buffer to establish the starting PTS value
+  endTime = 0;
+  sourceBuffer.segmentParser_.trigger('data', {
+    tags: {
+      videoTags: [makeFlvTag(19 * 1000, new Uint8Array(1))],
+      audioTags: [makeFlvTag(19 * 1000, new Uint8Array(1))]
+    }
+  });
+  timers.runAll();
+
+  sourceBuffer.appendBuffer(new Uint8Array(10));
+  timers.runAll();
+
+  // mock out a new segment of FLV tags, starting 10s after the
+  // starting PTS value
+  while (i--) {
+    tags_.unshift(
+      makeFlvTag((i * 1000) + (29 * 1000),
+        new Uint8Array([i])));
+  }
+  sourceBuffer.segmentParser_.trigger('data', {
+    tags: {
+      videoTags: tags_,
+      audioTags: tags_
```

Comment:
`videoTags` and `audioTags` should have easily identifiable tags so that we can be sure that the right ones are culled. Right now you can't be sure which 0-9 and which 7-9 there are represented in the assertion below.